### PR TITLE
Decompile start_menu

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/start_menu.c [hunk 1]
```diff
@@ -0,0 +1,1008 @@
...
+static EWRAM_DATA u8 sStartMenuCursorPos = 0;
+static EWRAM_DATA u8 sNumStartMenuItems = 0;
+static EWRAM_DATA u8 sStartMenuOrder[MAX_STARTMENU_ITEMS] = {};
+static EWRAM_DATA s8 sDrawStartMenuState[2] = {};
+static EWRAM_DATA u8 sSafariZoneStatsWindowId = 0;
+static ALIGNED(4) EWRAM_DATA u8 sSaveStatsWindowId = 0;
+
+static u8 (*sSaveDialogCB)(void);
+static u8 sSaveDialogDelay;
+static u8 sSaveDialogIsPrinting;
```
Comment (jiangzhengwenjz) [inline src/start_menu.c]: bool8

File: src/start_menu.c [hunk 2]
```diff
@@ -0,0 +1,1008 @@
...
+        return TRUE;
+    }
+    return FALSE;
+}
+
+static void StartMenu_PrepareForSave(void)
+{
+    save_serialize_map();
+    sSaveDialogCB = SaveDialogCB_PrintAskSaveText;
+    sSaveDialogIsPrinting = 0;
```
Comment (jiangzhengwenjz) [inline src/start_menu.c]: FALSE

File: src/start_menu.c [hunk 3]
```diff
@@ -0,0 +1,1008 @@
...
+}
+
+static u8 RunSaveDialogCB(void)
+{
+    if (RunTextPrinters_CheckPrinter0Active() == TRUE)
+        return 0;
+    sSaveDialogIsPrinting = 0;
```
Comment (jiangzhengwenjz) [inline src/start_menu.c]: FALSE

File: src/start_menu.c [hunk 4]
```diff
@@ -0,0 +1,1008 @@
...
+    StartMenu_PrepareForSave();
+    CreateTask(task50_save_game, 80);
+}
+
+void PrintSaveTextWithFollowupFunc(const u8 *str, bool8 (*savecb)(void))
+{
+    StringExpandPlaceholders(gStringVar4, str);
+    sub_80F7768(0, TRUE);
+    AddTextPrinterForMessage(TRUE);
+    sSaveDialogIsPrinting = 1;
```
Comment (jiangzhengwenjz) [inline src/start_menu.c]: TRUE

File: src/start_menu.c [hunk 5]
```diff
@@ -0,0 +1,1008 @@
...
+
+void Field_AskSaveTheGame(void)
+{
+    HelpSystem_BackupSomeVariable();
+    HelpSystem_SetSomeVariable2(12);
+    StartMenu_PrepareForSave();
+    CreateTask(task50_save_game, 80);
+}
+
+void PrintSaveTextWithFollowupFunc(const u8 *str, bool8 (*savecb)(void))
```
Comment (jiangzhengwenjz) [inline src/start_menu.c]: maybe `saveCb`is better?

File: src/quest_log.c [hunk 6]
```diff
@@ -109,7 +109,7 @@ EWRAM_DATA u8 gUnknown_203AF9A[64][2] = {{0}};
 EWRAM_DATA u16 gUnknown_203B01A = 0;
 EWRAM_DATA u16 gUnknown_203B01C = 0;
 EWRAM_DATA u16 gUnknown_203B01E = 0;
-EWRAM_DATA u8 gUnknown_203B020 = 0;
+EWRAM_DATA u8 sHelpMessageWindowId = 0;
```
Comment (jiangzhengwenjz) [inline src/quest_log.c]: static

File: src/quest_log.c [hunk 7]
```diff
@@ -2294,37 +2294,37 @@ void sub_8112E3C(u8 a0, struct UnkStruct_300201C * a1, u16 a2)
 
 const u16 gUnknown_84566A8[] = INCBIN_U16("data/graphics/unknown_84566a8.bin");
 
-const struct WindowTemplate gUnknown_8456928 = {
+const struct WindowTemplate sHelpMessageWindowTemplate = {
```
Comment (jiangzhengwenjz) [inline src/quest_log.c]: static

File: src/start_menu.c [hunk 8]
```diff
@@ -0,0 +1,1008 @@
...
+        data[0]++;
+        break;
+    case 1:
+        if (sStartMenuCallback() == TRUE)
+            DestroyTask(taskId);
+        break;
+    }
+}
+
+void ShowStartMenu(void)
```
Comment (jiangzhengwenjz) [inline src/start_menu.c]: This should be in include/start_menu.h

File: data/strings.s [hunk 9]
```diff
@@ -351,19 +351,19 @@ gUnknown_8416133:: @ 0x8416133
 gUnknown_8416146:: @ 0x8416146
 	.string "ポケモン$"
 
-gUnknown_841614B:: @ 841614B
+gText_SaveError_PleaseExchangeBackupMemory:: @ 841614B
 	.string "Save error.\pPlease exchange the\nbackup memory.$"
```
Comment (ultima-soul) [inline data/strings.s]: Possibly you can split this into separate lines for consistency with other strings?

**Pull request merged**